src/Functions/leftPadString.cpp

```diff
@@ -0,0 +1,205 @@
+            res_offsets.resize(size);
+
+            const size_t padstr_size = padstr.size();
+
+            ColumnString::Offset prev_offset = 0;
+            ColumnString::Offset res_prev_offset = 0;
+            for (size_t i = 0; i < size; ++i)
+            {
+                size_t data_length = offsets[i] - prev_offset - 1 /* zero terminator */;
+                if (data_length < length)
+                {
+                    for (size_t j = 0; j < length - data_length; ++j)
+                        res_data[res_prev_offset + j] = padstr[j % padstr_size];
+                    memcpy(&res_data[res_prev_offset + length - data_length], &data[prev_offset], data_length);
+                }
+                else
+                {
+                    for (size_t j = 0; j < length; ++j)
+                        res_data[res_prev_offset + j] = data[prev_offset + j];
+                }
+                res_data[res_prev_offset + length] = 0;
+                res_prev_offset += length + 1;
+                res_offsets[i] = res_prev_offset;
+            }
+        }
+
+        static void vectorFixed(
+            const ColumnFixedString::Chars & data,
+            const size_t n,
+            const size_t length,
+            const String & padstr,
+            ColumnFixedString::Chars & res_data)
+        {
+            const size_t padstr_size = padstr.size();
+            const size_t size = data.size() / n;
+            res_data.resize(length * size);
+            for (size_t i = 0; i < size; ++i)
+            {
+                if (length < n)
+                {
+                    memcpy(&res_data[i * length], &data[i * n], length);
+                }
+                else
+                {
+                    for (size_t j = 0; j < length - n; ++j)
+                        res_data[i * length + j] = padstr[j % padstr_size];
+                    memcpy(&res_data[i * length + length - n], &data[i * n], n);
+                }
+            }
+        }
+    };
+
+    class FunctionLeftPadString : public IFunction
+    {
+    public:
+        static constexpr auto name = "leftPadString";
+        static FunctionPtr create(const ContextPtr) { return std::make_shared<FunctionLeftPadString>(); }
+
+        String getName() const override { return name; }
+
+        bool isVariadic() const override { return true; }
+        size_t getNumberOfArguments() const override { return 0; }
+
+        bool useDefaultImplementationForConstants() const override { return true; }
+
+        DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+        {
+            size_t number_of_arguments = arguments.size();
+
+            if (number_of_arguments != 2 && number_of_arguments != 3)
+                throw Exception(
+                    ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH,
+                    "Number of arguments for function {} doesn't match: passed {}, should be 2 or 3",
```

Comment:
`lpad()` in MySQL always requires 3 arguments, why don't do the same?

Reply:
I was looking at oracledb

Reply:
ok